### PR TITLE
allow creation of service accounts

### PR DESCRIPTION
**Pull request by thecodeassassin**
new methods:

* cluster.CreateServiceAccount
* cluster.DeleteServiceAccount
* cluster.ListUserServiceAccounts

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -124,3 +136,67 @@ func (c *ClusterService) ListNodes(clusterName string) (*NodeList, error) {
...
+			&serviceAccount)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return serviceAccount, nil
+}
+
+func (n *ClusterService) GetServiceAccount(clusterName string, serviceAccountName string) (*ServiceAccount, error) {
```
Comment (Styren) [inline cluster.go:162]: Should we rename serviceAccountName to serviceAccountID? Since it's a UUID

File: cluster.go [hunk 2]
```diff
@@ -124,3 +136,67 @@ func (c *ClusterService) ListNodes(clusterName string) (*NodeList, error) {
...
+			&serviceAccount)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return serviceAccount, nil
+}
+
+func (n *ClusterService) DeleteServiceAccount(clusterName string, serviceAccountName string) error {
```
Comment (Styren) [inline cluster.go:177]: Same here

File: cluster.go [hunk 3]
```diff
@@ -124,3 +136,67 @@ func (c *ClusterService) ListNodes(clusterName string) (*NodeList, error) {
...
 }
+
+type ServiceAccount struct {
+	ID                          string `json:"id"`
+	KubeConfig                  string `json:"kubeConfig"`
+	ServiceAccountToken         string `json:"serviceAccountToken"`
+	ClusterCertificateAuthority string `json:"clusterCertificateAuthority"`
+}
+
+func (n *ClusterService) CreateServiceAccount(clusterName string) (*ServiceAccount, error) {
```
[the remark this answers is not in the document]
Comment (thecodeassassin) [reply cluster.go:147]: But not via the API correct?

Agree that `CreateServiceAccountForSelf` makes a lot of sense here.

[the remark this answers is not in the document]
Comment (Styren) [reply cluster.go]: It's possible, `rest/v1/cluster/%s/user-service-account/SUBJECT_ID` will create it on behalf of someone else

**Review by Styren — APPROVED** [review]

**Review by Styren — APPROVED** [review]
LGTM🚀

**Pull request merged**